crates/resolver/src/resolver.rs

```diff
@@ -49,6 +49,12 @@ impl<P> ResolverBuilder<P>
 where
     P: ConnectionProvider,
 {
+    /// Sets the [`ResolverOpts`] to be used by the resolver.
```

Comment:
I think this is a good idea. I agree the `options_mut()` API is a little idiosyncratic, though there is a reason for it: if you call `Resolver::builder()` you get the system configuration by default. This also influences the `ResolverOpts`. As such, I think the documentation for this method should at least explain that this overwrites any options set from the system configuration (`resolv.conf`/the registry).

Reply:
that's a great idea @djc. i've added some additional color about the defaults in 508a87b3. this notes that this method overwrites the system configuration, and points to (a) where to find parsing functions, and (b) the `options_mut()` as a related alternative.

Reply:
Thanks! I'll take this and will followup to tweak the new docs.